packages/lit-dev-content/site/docs/components/shadow-dom.md

```diff
@@ -205,37 +205,48 @@ render() {
 
 For more information, see [HTMLSlotElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement) on MDN.
 
-### @queryAssignedNodes decorator { #query-assigned-nodes }
+### @queryAssignedElements and @queryAssignedNodes decorator { #query-assigned-elements }
 
-The `@queryAssignedNodes` decorator converts a class property into a getter that returns all of the assigned nodes for a given slot in the component's shadow tree. The optional second boolean argument when true flattens the assigned nodes, meaning any assigned nodes that are slot elements are replaced with their assigned nodes. The optional third argument is a css selector which filters the results to matching elements.
+`@queryAssignedElements` and `@queryAssignedNodes` convert a class property into a getter that returns the result of calling
+[`HTMLSlot.assignedElements`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/assignedElements) or [`HTMLSlot.assignedNodes`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/assignedNodes) respectively on a given slot in the component's shadow tree.
+Use these to query the elements or nodes assigned to the given slot.
+
+Both accept an options object similar to what the native browser API accepts, with the addition of `slot`.
+Use `slot` to specify the slot to query, or leave undefined for the default slot.
+
+`@queryAssignedElements` also supports an optional `selector` option for filtering the returned elements by a CSS selector.
+
+Deciding which decorator to use depends on whether you want to query for text nodes assigned to the slot, or only element nodes. This decision is specific to your use case.
 
 <div class="alert alert-info">
 
 **Using decorators.** Decorators are a proposed JavaScript feature, so you’ll need to use a compiler like Babel or TypeScript to use decorators. See [Using decorators](/docs/components/decorators/) for details.
 
 </div>
 
-```js
-// First argument is the slot name
-// Second argument is `true` to flatten the assigned nodes.
-@queryAssignedNodes('header', true)
-_headerNodes;
-
-// If the first argument is absent or an empty string, list nodes for the default slot.
-@queryAssignedNodes()
-_defaultSlotNodes;
-```
+{% switchable-sample %}
```

Comment:
It seems maybe a bit weird to have a switchable sample here, because this section is explicitly about the decorator. So we talk about decorators, but then if you're in JS mode, you don't actually see any decorators in the example. The section above this section already shows how to write equivalent code without decorators.

So I think I'd actually revert to just showing the decorator code here.

Reply:
We originally had the JavaScript to show what is generated. Do you think it's useful to add a comment to the JavaScript sample to make it clear that it represents the generated logic from using the decorator? Splitting it back into two sections does make sense though. Thank you for commenting on this.

Reply:
Yeah I kinda like it in two sections.

Reply:
It's much better. Thank you!